### PR TITLE
Coverage is missing coverage reports from test tmp dirs

### PR DESCRIPTION
**Pull request by dmoody256**
This PR adds the use of the COVERAGE_FILE environment variable so that coverage reports from test run in the tmp dir will be saved after those tmp dirs get deleted.

This is a resubmit of this PR: https://github.com/SCons/scons/pull/36 that was never successfully added when the Github issues were attempted to be migrated from tigris.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -51,34 +51,43 @@ jobs:
...
       # in constantly changing coverage reports depending on the number of times
       # the JobTests.py had to run to pass
       # TODO: figure out how to cover JobTests.py
       # - n=0; while [[ $n -lt 10 ]]; do coverage run --rcfile=$PWD/.coveragerc runtest.py src/engine/SCons/JobTests.py && break; n=$((n+1)); done; if [ "$n" -gt "9" ]; then false; fi
       # exclude JobTest.py becuase we already ran that
         - echo "src/engine/SCons/JobTests.py" > exclude_jobtest
-      # also exclude this test since it overides the exit function which doesnt work with coverage 
-      # more info here: https://coverage.readthedocs.io/en/coverage-4.4.2/subprocess.html#
-      # TODO: figure out how to cover test/exitfns.py
-        - echo "test/exitfns.py" >> exclude_jobtest
```
Comment (bdbaddog) [inline .travis.yml]: Is this test no longer an issue?

File: test/Repository/M4.py [hunk 2]
```diff
@@ -36,6 +36,10 @@
 
 test = TestSCons.TestSCons()
 
+# TODO: figure out how to write the coverage data to the locked folder or maybe somewhere else
+if 'COVERAGE_PROCESS_START' in os.environ:
```
Comment (bdbaddog) [inline test/Repository/M4.py]: Can you move the test for running under coverage to QMTest/TestSCons.py such that you can use : if test.coverage_run():   Just to simplify any changes going forward?

**Comment by bdbaddog** [comment]
Thanks for re-submitting the pull request with updates post bug migration snafu.

**Comment by dmoody256** [comment]
I moved the skipping of that test from the travis.yml to the test itself,
where it checks for the coverage environment variable, to make travis.yml
more agnostic

**Comment by bdbaddog** [comment]
Looks good. Merging.

**Pull request merged**